src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -989,20 +1011,9 @@ void RealmCoordinator::run_async_notifiers()
             notifier->attach_to(new_notifier_transaction);
             notifier->add_required_change_info(new_notifier_change_info->current());
         }
-        new_notifier_change_info->advance_to_final(VersionID{});
-
-        // We want to advance the non-new notifiers to the same version as the
-        // new notifiers to avoid having to merge changes from any new
-        // transaction that happen immediately after this into the new notifier
-        // changes
-        version = new_notifier_transaction->get_version_of_current_transaction();
+        new_notifier_change_info->advance_to_final(version);
     }
     else {
-        // If we have no new notifiers we want to just advance to the latest
-        // version, but we have to pick a "latest" version while holding the
-        // notifier lock to avoid advancing over a transaction which should be
-        // skipped
-        version = m_db->get_version_id_of_latest_snapshot();
```

Comment:
Can we release `lock` earlier (as before 368f0c7e5eebc65d3326bb5a973df8aa99877fc5) now that we retrieved both the version and the skip version under it?

Reply:
`m_notifier_cv.notify_all()` is the remaining thing that's being guarded by the lock after where it was previously released. I think the code as it currently exists doesn't need to hold the mutex when notifying the CV, but I think doing so simplifies reasoning about the threading more than it complicates this function.

Reply:
Ok thanks for checking!